build-binutils.py

```diff
@@ -0,0 +1,221 @@
+                        it to this parameter. This can either be an absolute or relative path.
+                        """,
+                        type=str,
+                        default=os.path.join(root.as_posix(), "build"))
+    parser.add_argument("-t",
+                        "--targets",
+                        help="""
+                        The script can build binutils targeting arm-linux-gnueabi, aarch64-linux-gnu,
+                        powerpc-linux-gnu, powerpc64le-linux-gnu, and x86_64-linux-gnu.
+
+                        You can either pass the full target or just the first part (arm, aarch64, x86_64, etc)
+                        or all if you want to build all targets (which is the default). It will only add the
+                        target prefix if it is not for the host architecture.
+                        """,
+                        nargs="+")
+    return parser.parse_args()
+
+
+def create_targets(targets):
+    """
+    Generate a list of targets that can be passed to the binutils compile function
+    :param targets: A list of targets to convert to binutils target triples
+    :return: A list of target triples
+    """
+    targets_dict = {
+        "arm": "arm-linux-gnueabi",
+        "aarch64": "aarch64-linux-gnu",
+        "powerpc64le": "powerpc64le-linux-gnu",
+        "powerpc": "powerpc-linux-gnu",
+        "x86_64": "x86_64-linux-gnu"
+    }
+    targets_list = []
+
+    if len(targets) == 1 and targets[0] == "all":
+        targets_list = list(targets_dict.values())
+    else:
+        for target in targets:
+            if target == "host":
+                key = host_arch_target()
+            else:
+                key = target_arch(target)
+            targets_list.append(targets_dict[key])
+
+    return targets_list
+
+
+def cleanup(build_folder):
+    """
+    Cleanup the build directory
+    :param build_folder: Build directory
+    """
+    if build_folder.is_dir():
+        shutil.rmtree(build_folder.as_posix())
+    build_folder.mkdir(parents=True, exist_ok=True)
+
+
+def invoke_configure(build_folder, install_folder, root, target):
+    """
+    Invokes the configure script to generate a Makefile
+    :param build_folder: Build directory
+    :param install_folder: Directory to install binutils to
+    :param root: Working directory
+    :param target: Target to compile for
+    """
+    configure = [
+        root.joinpath(utils.current_binutils(), "configure").as_posix(),
+        '--prefix=' + install_folder.as_posix(),
+        '--enable-deterministic-archives', '--enable-gold',
+        '--enable-ld=default', '--enable-plugins', '--quiet',
+        'CFLAGS=-O2 -march=native -mtune=native',
+        'CXXFLAGS=-O2 -march=native -mtune=native'
+    ]
+    if "arm" in target or "aarch64" in target:
+        configure += [
+            '--disable-multilib', '--disable-nls',
+            '--program-prefix=' + target + '-', '--target=' + target,
```

Comment:
Consider using one of the formatting techniques here (i.e. the `%` operator or f-strings in 3.6+) and in the rest of the cases (because it is easier to understand what will be the final string using those and makes it easier to spot that formatting is happening).

Reply:
Is this what you are envisioning? https://github.com/ClangBuiltLinux/tc-build/commit/3d3b6c2d8395b21418b77c3c3276332cd9d22a01

Reply:
Yep!